tapeagents/finetune/rl/__init__.py

```diff
@@ -196,6 +201,17 @@ def update_advantages(dataset: Dataset, config: RLConfig) -> Dataset:
 
     """
     df = dataset.to_pandas()
+    
+    logging.info("Update Reward with Implicit KL")
+    def calculate_reward_with_implicit_kl(row):
```

Comment:
this function does not use any variables of the function in which it is located, I'd move it to the top-level. Inner functions are often not so great for readability.